services/src/main/java/org/keycloak/services/resources/admin/IdentityProvidersResource.java

```diff
@@ -134,6 +135,9 @@ public Response getIdentityProviders(@PathParam("provider_id") String providerId
         if (!(data.containsKey("providerId") && data.containsKey("fromUrl"))) {
             throw new BadRequestException();
         }
+        
+        ReservedCharValidator.validate((String)data.get("alias"));
```

Comment:
Should this be data.get("alias").toString()?

Reply:
@stianst I guess it depends on what you want to happen if somehow "alias" is not a string.  That would be an error.  With a cast you get ClassCastException.  With toString(), you get unpredictable results.